code/modules/clothing/masks/gasmask.dm

```diff
@@ -15,15 +15,20 @@
 	///List to keep track of each filter
 	var/list/gas_filters
 	///Is the filter allowed to spawn from the start?
-	var/starting_filter = TRUE
+	var/plasmaman_type = FALSE
 
 /obj/item/clothing/mask/gas/Initialize()
 	. = ..()
-	if(!max_filters || !starting_filter)
+	if(!max_filters)
 		return
+
+	var/filter_type = /obj/item/gas_filter
+	if(plasmaman_type)
+		filter_type = /obj/item/gas_filter/plasmaman
```

Comment:
Instead of having `plasmamen_type` a boolean, just have
`var/starting_filter_type = /obj/item/gas_filter`, and then use that.

Then override that var on subclasses.

Reply:
done